specific/cmo_stock/models/stock.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+from openerp.tools.translate import _
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    partner_id = fields.Many2one(
+        'res.partner',
+        default=lambda self: self._get_partner_id(),
+    )
+    default_operating_unit_id = fields.Many2one(
+        'operating.unit',
+        string='Default Operating Unit',
+        compute='_compute_default_operating_unit_id',
+    )
+
+    @api.model
+    def _get_partner_id(self):
+        user = self.env['res.users'].browse(self._uid)
+        return user.partner_id and user.partner_id.id or False
+
+    @api.multi
+    @api.depends('partner_id')
+    def _compute_default_operating_unit_id(self):
+        for picking in self:
+            User = picking.env['res.users']
+            picking.default_operating_unit_id = False
+            if picking.partner_id:
+                user = User.search([('partner_id', '=',
+                                     picking.partner_id.id)])
+                if user:
+                    picking.default_operating_unit_id = \
+                        user[0].default_operating_unit_id.id
+
+    @api.multi
+    def action_open_stock(self):
+        # Get product_id in each line
+        product_ids = []
+        for picking in self:
+            product_ids += picking.move_lines.mapped('product_id.id')
```

Comment:
I' can't mapped('product_id') because product_id is many to one, which reture result is

![image](https://user-images.githubusercontent.com/24691983/29164148-bef7ebf4-7de8-11e7-98f1-8b4b59e6f062.png)

and when I play on UI result is

![image](https://user-images.githubusercontent.com/24691983/29164192-df0b6010-7de8-11e7-92c9-5f87a2df234b.png)

which it's not correct
[Objective]
1. I want to list of product_id (not list of object product_id)
2. Use product_ids in searching products